cmake/linker_script/arm/linker.cmake

```diff
@@ -129,7 +129,7 @@ include(${COMMON_ZEPHYR_LINKER_DIR}/common-ram.cmake)
 #include(kobject.ld)
 
 if(NOT CONFIG_USERSPACE)
-  zephyr_linker_section(NAME .bss VMA RAM LMA FLASH TYPE BSS)
+  zephyr_linker_section(NAME .bss GROUP RAM_REGION TYPE BSS)
```

Comment:
As long as XIP is active, this is effectively no change, since `RAM_REGION` is `VMA RAM`/`LMA FLASH`; it's merely a cleanup and deduplication. Without XIP, `RAM_REGION` would be `VMA RAM`/`LMA RAM`, but I don't think it would actually cause the issue in #57590, since there's nothing in ROM that would create a huge hole between RAM and ROM.

Reply:
Right, the use of `GROUP` over the VMA / LMA use in f5eada5553467c732e7e5bf283e72b66d7be71a9 seems to be a better (safer) choice, and is also more aligned with the linker template in Zephyr.

Will carry out some extra tests before resolving this comment.

Reply:
ok, this seems ok, but found some other issues.